contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,333 @@
+     *  EVENT BASED PREDICTION MARKET DATA STRUCTURES  *
+     *************************************/
+
+    // Define the contract's constructor parameters as a struct to enable more variables to be specified.
+    struct ConstructorParams {
+        string pairName;
+        bytes32 priceIdentifier; // Price identifier, registered in the DVM for the long short pair.
+        IERC20 collateralToken; // Collateral token used to back LSP synthetics.
+        BinaryOptionLongShortPairFinancialProductLibrary financialProductLibrary; // Contract providing settlement payout logic.
+        bytes customAncillaryData; // Custom ancillary data to be passed along with the price request to the OO.
+        FinderInterface finder; // DVM finder to find other UMA ecosystem contracts.
+        address timerAddress; // Timer used to synchronize contract time in testing. Set to 0x000... in production.
+    }
+
+    bool public priceRequested;
+    bool public receivedSettlementPrice;
+
+    uint256 public expirationTimestamp;
+    string public pairName;
+    uint256 public collateralPerPair = 1 ether; // Amount of collateral a pair of tokens is always redeemable for.
+
+    // Number between 0 and 1e18 to allocate collateral between long & short tokens at redemption. 0 entitles each short
+    // to collateralPerPair and each long to 0. 1e18 makes each long worth collateralPerPair and short 0.
+    uint256 public expiryPercentLong;
+    bytes32 public priceIdentifier;
+
+    // Price returned from the Optimistic oracle at settlement time.
+    int256 public expiryPrice;
+
+    // External contract interfaces.
+    IERC20 public collateralToken;
+    ExpandedIERC20 public longToken;
+    ExpandedIERC20 public shortToken;
+    FinderInterface public finder;
+    BinaryOptionLongShortPairFinancialProductLibrary public financialProductLibrary;
+
+    // Optimistic oracle customization parameters.
+    bytes public customAncillaryData;
+    uint256 public proposerReward = 10 ether;
+    uint256 public optimisticOracleLivenessTime = 3600; // 1 hour
+    uint256 public optimisticOracleProposerBond = 500 ether;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event TokensCreated(address indexed sponsor, uint256 indexed collateralUsed, uint256 indexed tokensMinted);
+    event TokensRedeemed(address indexed sponsor, uint256 indexed collateralReturned, uint256 indexed tokensRedeemed);
+    event PositionSettled(address indexed sponsor, uint256 collateralReturned, uint256 longTokens, uint256 shortTokens);
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
+
+    modifier hasPrice() {
+        require(
+            _getOptimisticOracle().hasPrice(address(this), priceIdentifier, expirationTimestamp, customAncillaryData),
+            "Doesn't have price."
+        );
+        _;
```

Comment:
this is the kind of place I'd like to see it span over fewer lines. I know it seems silly...it's just that this repo is meant to be _as simple as posible_.